src/app/ipv/middleware.js

```diff
@@ -210,6 +211,40 @@ function checkJourneyAction(req) {
   }
 }
 
+// getCoiUpdateDetailsJourney determines the next journey based on the detailsToUpdate
+// field of the update-details page
+function getCoiUpdateDetailsJourney(detailsToUpdate) {
+  if (detailsToUpdate) {
+    // convert to array if its a string
+    if (typeof detailsToUpdate === "string") {
+      detailsToUpdate = [detailsToUpdate];
+    }
+    // cancel event
+    if (detailsToUpdate.includes("cancel")) {
+      return UPDATE_DETAILS_JOURNEY_TYPES.UPDATE_CANCEL;
+    }
+    // send to update-names-dob journey if dateOfBirth or both names selected
+    if (
+      detailsToUpdate.includes("dateOfBirth") ||
+      (detailsToUpdate.includes("givenNames") &&
+        detailsToUpdate.includes("lastName"))
+    ) {
+      return UPDATE_DETAILS_JOURNEY_TYPES.UPDATE_NAMES_DOB;
+    }
```

Comment:
What happens if the user selects given name and date of birth for example? Is that a valid selection?

Reply:
@RogVanGDS According to the diagram in https://govukverify.atlassian.net/browse/PYIC-5809 if DOB is selected with anything, it should  go to UPDATE_NAMES_DOB

Reply:
Ah ok, makes sense thanks 👍